pkg/identity/cache/cell/cell.go

```diff
@@ -120,6 +138,12 @@ func newIdentityAllocator(params identityAllocatorParams) identityAllocatorOut {
 		},
 	})
 
+	iao.updatePolicyMaps = job.NewTrigger()
+	jg := params.Registry.NewGroup(params.Health, job.WithMetrics(params.Metrics), job.WithLogger(params.Log))
+	jg.Add(job.Timer("id-alloc-update-policy-maps", iao.doUpdatePolicyMaps,
+		/* no interval, only on trigger */ 0, job.WithTrigger(iao.updatePolicyMaps)))
```

Comment:
aside: How does the batching work? I don't see a debouncer here so it seems like the job is run every time the trigger is hit. Is it just batched in that multiple calls to `UpdateIdentities()` are accumulated while `doUpdatePolicyMaps` propagates the accumulated changes into BPF? On average I'd expect the latter to be pretty quick so I would have figured there isn't much batching at all, but I guess in some cases with many map entries to update then it may take longer.

Reply:
`endpoint.ApplyPolicyMapChanges()` can also do a full Envoy recalculation + update + wait for ack, so it can take a very long time (100+ms in bad cases). So there is definite value in batching where possible.

You're right that there is no debouncing, just buffering. This is because, as you say, in the non-Envoy case updates are reliably very quick.

This basically mirrors the previous behavior, where endpoint regeneration requests are folded behind a queue but there is no debouncing.

The previous situation was pretty wasteful, a steady stream of identity allocations meant near-constant regenerations.